src/jvmMain/kotlin/io/github/fourlastor/editor/PreviewPane.kt

```diff
@@ -105,10 +108,47 @@ private data class GroupPreview(
 private data class ImagePreview(
     val transform: Transform,
     val image: ImageBitmap,
+    val frame: Frame,
 ) : EntityPreview() {
+
+    val Frame.maxNumberFrames: Int
+        get() {
+            return columns * rows
+        }
+    val Frame.frameNumberAdjusted: Int
+        get() {
+            return frameNumber % maxNumberFrames
+        }
+    val Frame.width: Int
+        get() {
+            return image.width / columns
+        }
+    val Frame.left: Int
+        get() {
+            return width * (frameNumberAdjusted % columns)
+        }
+    val Frame.right: Int
+        get() {
+            return left + width
+        }
+    val Frame.height: Int
+        get() {
+            return image.height / rows
+        }
+    val Frame.top: Int
+        get() {
+            return height * (frameNumberAdjusted / columns)
+        }
+    val Frame.bottom: Int
+        get() {
+            return top + height
+        }
```

Comment:
if we use these only here, we should make them `private`

Reply:
these might be useful functions for Frame to have, would it make sense to move them to the Frame class? if not I will make them private 👍

Reply:
actually it looks like there is an issue moving these to Frame due to the dependency on image - I will make these private for now then 👍